experimental/CliffordAlgebras/src/CliffordAlgebras.jl

```diff
@@ -0,0 +1,589 @@
+#  binary operators
+#
+################################################################################
+
+function Base.:+(x::CliffordAlgebraElem{T}, y::CliffordAlgebraElem{T}) where {T<:FieldElem}
+  @req parent(x) === parent(y) "The inputs must lie in the same Clifford algebra"
+  return parent(x)(coefficients(x) .+ coefficients(y))
+end
+
+Base.:-(x::CliffordAlgebraElem{T}, y::CliffordAlgebraElem{T}) where {T<:FieldElem} = x + -y
+
+function Base.:*(x::CliffordAlgebraElem{T}, y::CliffordAlgebraElem{T}) where {T<:FieldElem}
+  @req parent(x) === parent(y) "The inputs must lie in the same Clifford algebra"
+  xcoeffs, ycoeffs = copy(coefficients(x)), copy(coefficients(y))
+  return parent(x)(_mul_aux(xcoeffs, ycoeffs, gram_matrix(parent(x)), 1))
+end
+
+@doc raw"""
+    divexact(x::CliffordAlgebraElem, a::T) where {T<:RingElement} -> CliffordAlgebraElem
+
+Return the element `y` in the Clifford algebra containing $x$ such that $ay = x$,
+if it exists. Otherwise an error is raised.
+"""
+divexact(x::CliffordAlgebraElem, a::T) where {T<:RingElement} =
+  parent(x)(divexact.(coefficients(x), a))
+
+#divexact(x::CliffordAlgebraElem, a::T) where {T<:Number} = parent(x)(divexact.(coefficients(x), a))
+
+################################################################################
+#
+#  Equality and hash
+#
+################################################################################
+
+function Base.:(==)(x::CliffordAlgebraElem{T}, y::CliffordAlgebraElem{T}) where {T}
+  @req parent(x) === parent(y) "The inputs must lie in the same Clifford algebra"
+  return coefficients(x) == coefficients(y)
+end
+
+function Base.hash(x::CliffordAlgebraElem, h::UInt)
+  b = 0x1c4629b4de23b24c % UInt
+  h = hash(parent(x), h)
+  h = hash(coefficients(x), h)
+  return xor(h, b)
+end
+
+################################################################################
+#
+#  Graded parts
+#
+################################################################################
+
+@doc raw"""
+    even_part(x::CliffordAlgebraElem) -> CliffordAlgebraElem
+
+Return the projection of $x$ onto the even Clifford algebra
+"""
+even_part(x::CliffordAlgebraElem) = parent(x)(even_coefficients(x))
+
+@doc raw"""
+    odd_part(x::CliffordAlgebraElem) -> CliffordAlgebraElem
+
+Return the projection of $x$ onto the odd Clifford algebra.
+"""
+odd_part(x::CliffordAlgebraElem) = parent(x)(odd_coefficients(x))
+
+################################################################################
+#
+#  Auxillary functions
+#
+################################################################################
+
+function _set_even_odd_coefficients!(x::CliffordAlgebraElem)
+  x.even_coeffs = map(
+    y -> if sum(digits(y - 1; base=2, pad=_dim_qf(x.parent))) % 2 == 0
+      x.coeffs[y]
+    else
+      x.parent.base_ring()
+    end, 1:(x.parent.dim)
+  )
+  x.odd_coeffs = x.coeffs - x.even_coeffs
+  return x
+end
+
+function __can_convert_coefficients(coeff::Vector{R}, K::Field) where {R}
```

Comment:
(We need a function to do this. I have written such a function a dozen of times now.)